docs/design/projects/2023_member_code_enforcement.md

```diff
@@ -0,0 +1,47 @@
+# Member code enforcement
+
+[GitHub project](https://github.com/carpentries/amy/projects/15).
+
+In early 2024, the membership team planned to start enforcing that all preapproved IT applications and workshop requests should include a valid and active member code (a.k.a. registration code, group code)
+
+In mid-2023, the checkout requirements for Instructor Training were updated. As the relevant changes were made to AMY to support this update, the project also expanded to include improvements to the training progress view for trainees.
+
+## Feature flag
+
+The `ENFORCE_MEMBER_CODES` [feature flag](../feature_flags.md) controls whether validity checks are performed on member codes. If set to `True` in the Django admin panel, validity checks will be performed on all submissions. If set to `False`, no checks will be performed (i.e. all codes will be accepted). The value of the flag does **not** affect whether the `member_code` field is displayed on forms or in views.
```

Comment:
Need to switch the feature flag `ENFORCE_MEMBER_CODES` on before release 4.3.

Reply:
Done: https://github.com/carpentries/amy/commit/506a2fe544436d48389ae2e01295dae34ec24b41